transmart-batch/src/main/groovy/org/transmartproject/batch/facts/DummyFactGenerator.groovy

```diff
@@ -50,15 +64,39 @@ class DummyFactGenerator extends AbstractItemCountingItemStreamItemReader<Clinic
         ConceptNode concept = mapper[row]
         assert concept != null
 
+        TrialVisit trialVisit = getTrialVisit()
+
         ClinicalFactsRowSet rowSet = new ClinicalFactsRowSet(
                 studyId: studyId,
-                patient: patientSet[row.subjectId])
-
-        rowSet.addValue(concept, null, concept.name)
+                patient: patientSet[row.subjectId],
+                trialVisit: trialVisit
+        )
+
+        def sample_id = row.sampleCd
+        if (sample_id) {
+            // add text value with subject sample mapping
+            rowSet.addValue(concept, null, sample_id)
+
+            List<Map> result = getAssayIds(sample_id)
+
+            if (result.empty) {
+                return
+            }
+            // add modifier for each assay_id
+            String modifier = 'TRANSMART:HIGHDIM:' + platform.markerType.toUpperCase()
+            for (int i=0; i<result.size(); i++){
+                rowSet.instanceNum = i+1
```

Comment:
it does not look right. `instance_id` should be unique for each row. Pass it maybe to `addValue` method?